pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -96,6 +96,8 @@ public RealtimeSegmentDataManager(final RealtimeSegmentZKMetadata segmentMetadat
       RealtimeTableDataManager realtimeResourceManager, final String resourceDataDir, final ReadMode mode,
       final Schema schema, final ServerMetrics serverMetrics) throws Exception {
     super();
+    final String segmentVersionStr = tableConfig.getIndexingConfig().getSegmentFormatVersion();
```

Comment:
Currently the property is not there, so it is loading correctly for Realtime segments